src/frontend/web_application/src/components/ContactDetails/index.jsx

```diff
@@ -225,19 +302,86 @@ class ContactDetails extends Component {
     );
   }
 
-  renderContactDetailsForm() {
+  renderFormSelector = () => {
     const { __, contact } = this.props;
+
+    const emailOption = {
+      name: __('contact.form-selector.email_form.label'),
+      obj: (<EmailForm onSubmit={this.makeHandleAddContactDetail('emails')} __={__} />),
+    };
+
+    const phoneOption = {
+      name: __('contact.form-selector.phone_form.label'),
+      obj: (<PhoneForm onSubmit={this.makeHandleAddContactDetail('phones')} __={__} />),
+    };
+
+    const imOption = {
+      name: __('contact.form-selector.im_form.label'),
+      obj: (<ImForm onSubmit={this.makeHandleAddContactDetail('ims')} __={__} />),
+    };
+
+    const addressOption = {
+      name: __('contact.form-selector.address_form.label'),
+      obj: (<AddressForm onSubmit={this.makeHandleAddContactDetail('addresses')} __={__} />),
+    };
+
+    const formsOptions = [
```

Comment:
I'm not a big fan of this implem, the nodes for each option are created but unused if the contact property does exist. Hmm well, I don't think it has a big impact on the performances.

But it works so I'm good with it for now, I may refactor some of this when connecting api